### PR TITLE
added redirect_uri parameter to authentication call (including lib folder)

### PR DESCRIPTION
**Pull request by mostley**
github oauth supports different sub paths of the official callback: https://developer.github.com/v3/oauth/

### CODE REVIEW

File: src/octonode/auth.coffee [hunk 1]
```diff
@@ -74,6 +74,7 @@ auth = module.exports =
         uri+= "?client_id=#{@options.id}"
         uri+= "&state=#{randomstring.generate()}"
         uri+= "&scope=#{scopes.join(',')}"
+        if @options.redirect_uri then uri+= "&redirect_uri=#{@options.redirect_uri}"
```
Comment (pksunkara) [inline src/octonode/auth.coffee]: Can you add `return uri` line in here? If you see the generated `.js` file, it looks like coffee-script is introducing a bug by doing `return uri` inside the `if` loop.

**Comment by mostley** [comment]
done. Very strange behaviour and nice catch on your end.

**Comment by pksunkara** [comment]
Thanks for the PR.

**Comment by mostley** [comment]
can you bump the version and republish?

**Pull request merged**